File: FirClient/Assets/Scripts/Lua/Data/Tables/GlobalConfigTable.lua
local GlobalConfigTable = class("GlobalConfigTable")

function GlobalConfigTable:Initialize()
    self.Items = {
        ['CommonWhite'] = {id = 'CommonWhite', value = '250_250_250_255'},
        ['CommonGreen'] = {id = 'CommonGreen', value = '0_246_0_255'},
        ['CommonRed'] = {id = 'CommonRed', value = '250_17_17_255'}
	}
end

function GlobalConfigTable:GetItems()
    return self.Items
end

function GlobalConfigTable:GetItemByKey(key)
    return self.Items[key]
end

return GlobalConfigTable

File: FirClient/Assets/Scripts/Data/Tables/GlobalConfigTable.cs
using System.Collections.Generic;

namespace FirCommon.Data
{
	public class GlobalConfigTable
	{
		public string name;		
		public List<GlobalConfigTableItem> items = new List<GlobalConfigTableItem>();
		
		private Dictionary<string, GlobalConfigTableItem> dics = null;

		public void Initialize()
		{
			dics = new Dictionary<string, GlobalConfigTableItem>();
			foreach (GlobalConfigTableItem item in items)
			{
				dics.Add(item.id, item);
			}
		}

		public List<GlobalConfigTableItem> GetItems()
		{
			return items;
		}

		public void AddItem(GlobalConfigTableItem item)
		{
			items.Add(item);
		}

		public GlobalConfigTableItem GetItemByKey(string key)
		{
			GlobalConfigTableItem item = null;
			if (dics.ContainsKey(key))
			{
				dics.TryGetValue(key, out item);
			}
			return item;
		}
	}

	public class GlobalConfigTableItem
	{
    	public string id;
    	public string value;
	}
}